include/secp256k1_frost.h

```diff
@@ -335,6 +345,135 @@ SECP256K1_API int secp256k1_frost_nonce_gen(
     const unsigned char *extra_input32
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
 
+/** Takes the public nonces of all signers and computes a session that is
+ * required for signing and verification of partial signatures. The participant
+ * IDs can be sorted before combining, but the corresponding pubnonces must be
+ * resorted as well. All signers must use the same sorting of pubnonces,
+ * otherwise signing will fail.
+ *
+ *  Returns: 0 if the arguments are invalid or if some signer sent invalid
+ *           pubnonces, 1 otherwise
+ *  Args:          ctx: pointer to a context object
+ *  Out:       session: pointer to a struct to store the session
+ *  In:      pubnonces: array of pointers to public nonces sent by the signers
+ *         n_pubnonces: number of elements in the pubnonces array. Must be
+ *                      greater than 0.
+ *               msg32: the 32-byte message to sign
+ *              agg_pk: the FROST-aggregated public key
+ *               my_id: the ID of the participant who will use the session for
+ *                      signing
+*                  ids: array of the IDs of the signers
+ *         tweak_cache: pointer to frost_tweak_cache struct (can be NULL)
+ *             adaptor: optional pointer to an adaptor point encoded as a
+ *                      public key if this signing session is part of an
+ *                      adaptor signature protocol (can be NULL)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_frost_nonce_process(
+    const secp256k1_context *ctx,
+    secp256k1_frost_session *session,
+    const secp256k1_frost_pubnonce * const *pubnonces,
+    size_t n_pubnonces,
+    const unsigned char *msg32,
+    const secp256k1_xonly_pubkey *agg_pk,
+    size_t my_id,
+    const size_t *ids,
```

Comment:
In the documentation, we should probably mention that [1] the `ids` array must be unique, and [2] `my_id` must be included in `ids`. Otherwise, Lagrange interpolation will return incorrect values.

We could also implement `ARG_CHECK`s for these, but that might be a lot of work.

Reply:
In BIP, the _DeriveInterpolatingValue_ function fails if both these conditions aren't met.